src/charm.py

```diff
@@ -635,13 +610,19 @@ def _wp_generate_password(length=64):
             characters = string.ascii_letters + "!@#$%^&*()" + "-_ []{}<>~`+=,.;:/?|"
             return "".join(secrets.choice(characters) for _ in range(length))
 
-        return {
+        wp_secrets = {
             field: _wp_generate_password()
             for field in self._wordpress_secret_key_fields()
         }
+        wp_secrets["default_admin_password"] = secrets.token_urlsafe(32)
+        return wp_secrets
 
     def _replica_relation_data(self):
-        return self.model.get_relation("wordpress_replica").data[self.app]
+        relation = self.model.get_relation("wordpress-replica")
+        if relation is None:
+            return {}
```

Comment:
Looking through how the return value of `_replica_relation_data` is used, it seems that in many cases the keys that are expected are the same as the secret keys generated by the `_wordpress_secret_key_fields` function, wouldn't returning an empty dictionary here just lead to `KeyError` in all those functions? Maybe raising an error here is more appropriate?

Reply:
Updated, thanks!